feat(hardware/servo): implement a basic servo wrapper class

Baconing opened the pull request:
Implements a basic wrapper class for [Servo](https://github.wpilib.org/allwpilib/docs/development/java/edu/wpi/first/wpilibj/Servo.html).

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
	- Introduced a new `Servo` class to enhance control over servo motors, including speed and angle adjustments, position tracking.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
feat(hardware/servo): implement a basic servo wrapper class

Baconing commented on src/main/java/frc/robot/hardware/servo/Servo.java:
```diff
@@ -0,0 +1,86 @@
+     * Toggles the servo enabled status.
+     * @see edu.wpi.first.wpilibj.Servo#setDisabled()
+     */
+    public void toggle() {
+        servo.setDisabled();
```
I initially thought this too, but when reading the [documentation](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/Servo.html) of the Servo class, the [setDisabled()](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/PWM.html#setDisabled()) method actually toggles the enabled status.

However, the language used is a bit confusing, and I'm not sure if that's what's actually intended.
> Temporarily disables the PWM output. The next set call will re-enable the output.

So I'll probably wait to get a second pair of ears and see what they think. It might be worth changing the name of the function depending on what the setDisabled() function actually does.

Trip-kun replied:
setDisabled() disables the Servo until the following set() call, per WPILib docs

Remove Inversion, there is no need for that here.

Baconing commented on src/main/java/frc/robot/hardware/servo/Servo.java:
```diff
@@ -0,0 +1,86 @@
+     * @param angle the angle to set the servo to
+     * @see edu.wpi.first.wpilibj.Servo#setAngle(double)
+     */
+    public void setAngle(double angle) {
+        servo.setAngle(angle);
```
I am considering removing the inverted state all together, as I'm not sure if it should be considered when using servos. I only implemented it because the Motor classes have it, and I wanted to keep it fairly similar to them.

Baconing commented on src/main/java/frc/robot/hardware/servo/Servo.java:
```diff
@@ -0,0 +1,86 @@
+     * If the servo is inverted, the speed will be set to 1 - speed.
+     * @param speed the speed to set the servo to
+     */
+    public void set(double speed) {
+        if (inverted) {
+            servo.set(1 - speed);
+        } else {
+            servo.set(speed);
+        }
```
This is a good suggestion, however I am not sure if `speed` is the best variable name here, I think `position` is better. I am not sure why I called it speed as even WPILib refers to it as `value` and `position`, might've been done by GitHub Copilot and I glanced over this part.

Commit:
feat/fix(hardware/servo): make changes as specified in pr review

pr review: https://github.com/frc5183/robot/pull/16#pullrequestreview-1872506377

Trip-kun reviewed (approved):
Looks good to me
